### ISSUE TITLE
danger-swiftlint 0.11.0 is throwing false positives for delegate protocol class rule 

### ISSUE THREAD

**Issue by dubemike** [ashfurrow/danger-ruby-swiftlint#75]
<img width="687" alt="screen shot 2017-11-23 at 11 41 51" src="https://user-images.githubusercontent.com/6419590/33166814-4d405db2-d044-11e7-846a-4674c18aafed.png">

and you can see that line 104:

<img width="424" alt="screen shot 2017-11-23 at 11 42 59" src="https://user-images.githubusercontent.com/6419590/33166872-6b29d466-d044-11e7-8f26-c91ec16a382e.png">

running swiftlint locally won't throw this warning. any idea what could be happening here?

**Comment by ashfurrow** [comment]
Hmm, it sounds like the danger plugin isn't using the same `.swiftlint.yml` as when you run locally. Could you check your usage [against the docs](https://github.com/ashfurrow/danger-swiftlint#usage)? Thanks!

**Comment by kaybutter** [comment]
@dubemike  I had the same problem. The swiftlint binary that came with the gem was not up to date in my case. Setting the `binary_path` to my own swiftlint installation fixed that for me :)

**Comment by ajfigueroa** [comment]
It looks like this issue has been resolved in Swiftlint version [0.23.0](https://github.com/realm/SwiftLint/blob/master/CHANGELOG.md#0230-permanent-press-cycle).

Would the fix involve updating the `SWIFTLINT_VERSION` in [version.rb](https://github.com/ashfurrow/danger-swiftlint/blob/0f4cd70e8b71139d2784521de86acdb5e99b6bee/lib/version.rb#L5)? Would this likely be a breaking change?

**Comment by ashfurrow** [comment]
That makes sense to me. Since we're not upgrading the Swiftlint version to a new major version, I don't think this is a breaking change. Anyone up to test locally and send a PR?

**Comment by ajfigueroa** [comment]
There are a couple of breaking changes with SwiftLint from the current version `0.20.1` outlined from their [Changelog](https://github.com/realm/SwiftLint/blob/master/CHANGELOG.md) below:

| SwiftLint Version        | Breaking Change           |
|:-------------:|:-------------| 
| 0.21.0     | Xcode 8.3 or later and Swift 3.1 or later are required to build. | 
| 0.22.0      | Nested configurations will now be merged with parent configurations rather than replace them outright.  | 
| 0.23.0 & 0.23.1 | None    |
| 0.24.0 | SwiftLint now requires Xcode 9 and Swift 3.2+ to build and Remove SwiftExpressionKind.other. |

What do you all think? Since the issue has been resolved with `0.23.0`, it'd be nice to get there at least.
I imagine not many people are still using Xcode 8.3. I'm unsure about the nested configurations as I don't use them myself.

**Comment by thii** [comment]
@ajfigueroa Those changes are for building SwiftLint. This plugin use the prebuilt SwiftLint binary so it doesn't matter which Xcode people are using. You can use the `SWIFTLINT_VERSION` environment variable to override the default version, or you can specify a path to your SwiftLint binary.

```
swiftlint.binary_path = '/path/to/swiftlint'
```

**Comment by ajfigueroa** [comment]
@thii Oh, that's neat! My post can be disregarded then.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by omirho** [comment]
Swiftlint version was updated in the plugin. Closing this.
